lib/src/migrators/module.dart

```diff
@@ -738,110 +739,173 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
+  /// `@forward` rule from an import-only file that does not forward its
+  /// corresponding regular file. This allows imports of import-only files that
+  /// redirect to a different path to be migrated in-place.
+  List<String> _migrateImportToRules(String ruleUrl, FileSpan context) {
+    var tuple = _migrateImportCommon(ruleUrl, context);
+    var canonicalUrl = tuple.item1;
+    var config = tuple.item2;
+    var forwardForConfig = tuple.item3;
+
+    var asClause = '';
+    var defaultNamespace = namespaceForPath(ruleUrl);
+    // If a member from this dependency is actually referenced, it should
+    // already have a namespace from [_determineNamespaces], so we just use
+    // a simple number suffix to resolve conflicts at this point.
+    var namespace = _namespaces.putIfAbsent(canonicalUrl,
+        () => _incrementUntilAvailable(defaultNamespace, _namespaces));
+    if (namespace != defaultNamespace) asClause = ' as $namespace';
+
+    var quote = context.text.startsWith("'") ? "'" : '"';
+    var quotedUrl = '$quote$ruleUrl$quote';
+    var withClause = config == null ? '' : ' with $config';
+    var rules = <String>[];
+    if (forwardForConfig != null) {
+      rules.add('@forward $quotedUrl show $forwardForConfig$withClause');
+      // We consume the `with` clause if it exists so that it doesn't get
+      // duplicated on the `@use` rule.
+      withClause = '';
+    }
+    var normalForwardRules = _upstreamStylesheets.isEmpty
+        ? _makeForwardRules(canonicalUrl, quotedUrl)
+        : null;
+    if ((rules.isEmpty && normalForwardRules == null) ||
+        withClause.isNotEmpty ||
+        references.anyMemberReferenced(canonicalUrl, currentUrl)) {
+      _usedUrls.add(canonicalUrl);
+      rules.add('@use $quotedUrl$asClause$withClause');
+    }
+    if (normalForwardRules != null) rules.addAll(normalForwardRules);
+    return rules;
+  }
+
+  /// Similar to [_migrateImportToRules], but instead migrates an import to a
+  /// single `meta.load-css` call, or errors if this isn't allowed.
+  ///
+  /// This is used for migrating `@import` rules that are nested or appear after
+  /// some other rules.
+  String _migrateImportToLoadCss(String ruleUrl, FileSpan context) {
+    _unreferencable = UnreferencableMembers(_unreferencable);
+    for (var declaration in references.allDeclarations) {
+      if (declaration.sourceUrl != currentUrl) continue;
+      _unreferencable.add(declaration, UnreferencableType.fromImporter);
     }
+
+    var tuple = _migrateImportCommon(ruleUrl, context);
+    var canonicalUrl = tuple.item1;
+    var config = tuple.item2;
+    if (tuple.item3 != null) {
+      throw MigrationSourceSpanException(
+          "This declaration attempts to override a default value in an "
+          "indirect, nested import of ${p.prettyUri(canonicalUrl)}, which is "
+          "not possible in the module system.",
+          _configuredVariables.last.member.span);
```

Comment:
It looks like this text is duplicated on line 933. I that intentional? Are there two different ways of getting this error? If so, should you add an extra test case for the other way of getting it? Should you add a shared function to throw the error so you don't have to duplicate the text?

Reply:
Yeah, there's only supposed to be one. I factored this out into `_migrateImportToLoadCss` but forgot to remove it from the common logic. Fixed.